llm/gpu_darwin.go

```diff
@@ -0,0 +1,19 @@
+//go:build darwin
+
+package llm
+
+import (
+	"github.com/jmorganca/ollama/api"
+)
+
+// CheckVRAM returns the free VRAM in bytes on Linux machines with NVIDIA GPUs
+func CheckVRAM() (int64, error) {
+	// TODO - assume metal, and return free memory?
```

Comment:
I think this comment is right – there's till a few crashes folks will run into if they don't have enough free memory (and it doesn't seem like memory paging works).

Reply:
(Can solve this in a follow up)